packages/common/src/request.ts

```diff
@@ -17,34 +17,54 @@ export class RemoteServerError extends Error {
 }
 
 /**
+ * ```js
+ * import { hubApiRequest } from "@esri/hub-common";
+ * //
+ * hubApiRequest(
+ *   "/datasets",
+ *   requestOptions
+ * })
+ *   .then(response);
+ * ```
  * make a request to the Hub API
- * @param url URL to request
+ * @param route API route
  * @param requestOptions request options
  */
-export function hubRequest(url: string, requestOptions?: IHubRequestOptions) {
-  // TODO: use request() under the hood?
-  // requestOptions.params.f = null;
-  // return request(url, requestOptions);
-  // TODO: cast to JSONAPI document?
-  // .then(response => {
-  //   const { data, meta } = response;
-  //   return { data, meta } as Document;
-  // });
-  // TODO: base on requestOptions
-  const fetchFn = /* requestOptions.fetch || */ fetch;
-  return fetchFn(url, {
-    method: "POST", // TODO: get from requestOptions?
-    headers: {
-      // TODO: base on request requestOptions?
-      "Content-Type": "application/json"
-    }
-    // TODO: base on requestOptions.params?
-    // body: JSON.stringify(body)
-  }).then(resp => {
-    if (resp.ok) {
-      return resp.json();
-    } else {
-      throw new RemoteServerError(resp.statusText, url, resp.status);
-    }
+export function hubApiRequest(
+  route: string,
+  requestOptions?: IHubRequestOptions
+) {
+  // merge in default request options
+  const options: IHubRequestOptions = {
+    ...{
+      hubApiUrl: "https://opendata.arcgis.com/api/v3/",
+      httpMethod: "POST",
```

Comment:
Why is `POST` the default method?

Reply:
That's what we did in REST JS, b/c most of the requests made by the lib were POST (CUD vs R), and all the ArcGIS REST API "query" end points accept POST as well as GET.

I'd love your insight re: the Hub API. If you think GET is the better default, we can roll w/ that

Reply:
That's a tough one then.  I'd venture that most of the requests going to Hub API are going to be GETs - check out the API docs here: https://github.com/ArcGIS/hub-indexer/blob/master/docs/api.md.  Also, since node-fetch's default is `GET` I generally assume that it is the default of request methods.  If it were up to me I would default to GET.